tla/issue.md

```diff
@@ -0,0 +1,150 @@
+6. The backup (at view 0) replica 2 collects `M` prepare messages (from itself and replicas 0, 1) and broadcasts the `Commit` message for view 0.
+7. The backup (at view 0) replica 3 decides to change its view (possible on timeout) and sends the `ChangeView` message.
+8. The primary (at view 0) replica 0 collects `M` `ChangeView` messages (from itself and replicas 1, 3) and changes its view to 1.
+9. The backup (at view 0) replica 1 collects `M` `ChangeView` messages (from itself and replicas 0, 3) and changes its view to 1.
+10. The primary (at view 1) replica 1 sends the `PrepareRequest` message.
+11. The backup (at view 1) replica 0 receives the `PrepareResuest` of view 1 and sends the `PrepareResponse`.
+12. The backup (at view 1) replica 0 decides to change its view (possible on timeout) and sends the `ChangeView` message.
+13. The primary (at view 1) replica 1 decides to change its view (possible on timeout) and sends the `ChangeView` message.
+14. The backup (at view 0) replica 3 collects `M` `ChangeView` messages (from itself and replicas 0, 1) and changes its view to 1.
+15. The backup (at view 1) replica 3 receives `PrepareRequest` of view 1 and broadcasts its `PrepareResponse`.
+16. The backup (at view 1) replica 3 collects `M` prepare message and broadcasts the `Commit` message for view 1.
+17. The rest of undelivered messages eventually reaches their receivers, but it doesn't change the node's states.
+
+Here's the [TLC error trace](./base_deadlock1_dl.txt) attached.
+
+</details>
+
+After the described sequence of steps we end up in the following situation:
+ 
+| Replica | View | State                                                          |
+|---------|------|----------------------------------------------------------------|
+| 0       | 1    | `ChangeView` sent, in the process of changing view from 1 to 2 |
+| 1       | 1    | `ChangeView` sent, in the process of changing view from 1 to 2 |
+| 2       | 0    | `Commit` sent, waiting for the rest nodes to commit at view 0  |
+| 3       | 1    | `Commit` sent, waiting for the rest nodes to commit at view 1  |
+
+So we have the replica 2 stuck at the view 0 without possibility to exit from the commit
+stage and without possibility to collect more `Commit` messages from other replicas.
+We also have replica 3 stuck at the view 1 with the same problem. And finally, we have
+replicas 0 and 1 entered the view changing stage and not being able either to commit (as
+there's only `F` nodes that have been committed at the view 1) or to change view (as the
+replica 2 can't send its `ChangeView` from the commit stage).
+
+This liveness lock happens because the outcome of the subsequent consensus round (either
+commit or do change view) completely depends on the message receiving order.
+
+It should be noted that this kind of deadlock is possible in the 7-nodes network scenario
+(TODO: run simple model with 7 good nodes). Moreover, we've faced with exactly this kind
```

Comment:
I doubt 4-7 should be mentioned, it's the same protocol with the same basic behavior.